pandas/core/internals.py

```diff
@@ -5182,7 +5182,15 @@ def concatenate_block_managers(mgrs_indexers, axes, concat_axis, copy):
 
     for placement, join_units in concat_plan:
 
-        if is_uniform_join_units(join_units):
+        if len(join_units) == 1 and not join_units[0].indexers:
+            b = join_units[0].block
+            values = b.values
+            if copy:  # and values.base is not None:
```

Comment:
@jreback I first had this with `and values.base is not None:` but that failed one test in `merge`. But I copied that logic from `concatenate_join_units`:

https://github.com/jorisvandenbossche/pandas/blob/2125ac09726cfc720def79e836398ed04ecae04a/pandas/core/internals.py#L5332

Not sure why there it is needed, but here it fails.